themes/default/content/what-is/what-is-serverless.md

```diff
@@ -0,0 +1,112 @@
+
+1. **Choose the right trigger**: When dealing with serverless functions you should choose the trigger that best suits your use case. Common triggers include HTTP requests, message queues, database changes, and timers. In our application, we used an HTTP trigger for our two functions.
+
+    ``` go
+    // The 'date' function that returns the current date
+	mux.HandleFunc("/api/date", func(w http.ResponseWriter, r *http.Request) {
+        ...
+	})
+
+    // The 'weather' function that returns the weather
+	mux.HandleFunc("/api/weather", func(w http.ResponseWriter, r *http.Request) {
+        ...
+	})
+    ```
+
+1. **Avoid making direct cross-service references**: In your serverless architecture, there are times when functions depend on other functions. This typically happens a lot in distributed systems. For example in an e-commerce service, the checkout function could depend on the payment function. However, it's advised that services avoid making direct references to one another in order to prevent hard coupling that can cause issues later. A significant amount of reworking may be required if direct references for a service need to be changed. Using message services, such as queues or topics, to facilitate service-to-service communication is a good way to handle this. This decoupling enhances flexibility, scalability, and resilience, as well as simplifies testing and debugging. It also promotes security by allowing access control. When you use a more decoupled architecture, serverless systems can evolve independently and adapt to changing requirements, making them more robust and maintainable over time.
+
+## Conclusion
+
+Serverless architectures empower developers to build scalable, cost-effective applications without the burden of managing servers. By understanding its components, associated cloud services, and benefits, technical professionals can leverage serverless to streamline development processes and create more agile, responsive applications in the cloud.
+
+Pulumi's code-driven approach to infrastructure as code aligns well with serverless architectures, allowing developers to use the same tools for writing their serverless functions and their infrastructure definitions. To learn more about using Pulumi to deploy serverless applications, check out [this blog post](https://www.pulumi.com/blog/easy-serverless-apps-and-infrastructure-real-events-real-code/).
```

Comment:
Seems like these templates are a better destination to start than that blog post which is getting pretty old

https://www.pulumi.com/templates/serverless-application/

Reply:
I've updated the text to point to the serverless architecture templates instead of the blog post.